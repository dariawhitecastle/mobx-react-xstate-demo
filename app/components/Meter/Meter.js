/* eslint-disable react/prop-types */
/**
 * Meter
 */

import React from 'react'
import { Meter } from 'grommet'
import { doc } from './Meter.doc'
import helpers from '../../utils/helpers'

function createMeter({
  a11yTitle,
  alignSelf,
  background,
  gridArea,
  margin,
  round,
  size,
  thickness,
  values,
  ...rest
}) {
  return (
    <Meter
      a11yTitle={a11yTitle}
      alignSelf={alignSelf}
      background={background}
      gridArea={gridArea}
      margin={margin}
      round={round}
      size={size}
      thickness={thickness}
      values={values}
      {...rest}
    />
  )
}

export default helpers.createWithDoc({
  docFunction: doc,
  component: createMeter,
})
