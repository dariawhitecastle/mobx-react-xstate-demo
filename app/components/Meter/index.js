export { default as Meter } from './Meter'
