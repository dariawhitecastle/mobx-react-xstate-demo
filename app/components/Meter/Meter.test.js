import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import { Grommet } from 'grommet'
import Meter from './Meter'

const VALUES = [{ value: 20, label: 'twenty', onHover: () => {} }]

describe('Meter', () => {
  test('default', () => {
    const component = renderer.create(
      <Grommet>
        <Meter />
      </Grommet>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('basic', () => {
    const component = renderer.create(
      <Grommet>
        <Meter values={VALUES} />
      </Grommet>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('many values', () => {
    const component = renderer.create(
      <Grommet>
        <Meter
          values={[
            { value: 5 },
            { value: 5 },
            { value: 5 },
            { value: 5 },
            { value: 5 },
            { value: 5 },
            { value: 5 },
            { value: 5 },
          ]}
        />
      </Grommet>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('type', () => {
    const component = renderer.create(
      <Grommet>
        <Meter type="bar" values={VALUES} />
        <Meter type="circle" values={VALUES} />
      </Grommet>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('size', () => {
    const component = renderer.create(
      <Grommet>
        <Meter size="xsmall" values={VALUES} />
        <Meter size="small" values={VALUES} />
        <Meter size="medium" values={VALUES} />
        <Meter size="large" values={VALUES} />
        <Meter size="xlarge" values={VALUES} />
        <Meter size="24px" values={VALUES} />
        <Meter type="circle" size="xsmall" values={VALUES} />
        <Meter type="circle" size="small" values={VALUES} />
        <Meter type="circle" size="medium" values={VALUES} />
        <Meter type="circle" size="large" values={VALUES} />
        <Meter type="circle" size="xlarge" values={VALUES} />
        <Meter type="circle" size="55px" values={VALUES} />
      </Grommet>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('thickness', () => {
    const component = renderer.create(
      <Grommet>
        <Meter thickness="xsmall" values={VALUES} />
        <Meter thickness="small" values={VALUES} />
        <Meter thickness="medium" values={VALUES} />
        <Meter thickness="large" values={VALUES} />
        <Meter thickness="xlarge" values={VALUES} />
        <Meter thickness="55px" values={VALUES} />
        <Meter type="circle" thickness="xsmall" values={VALUES} />
        <Meter type="circle" thickness="small" values={VALUES} />
        <Meter type="circle" thickness="medium" values={VALUES} />
        <Meter type="circle" thickness="large" values={VALUES} />
        <Meter type="circle" thickness="xlarge" values={VALUES} />
        <Meter type="circle" thickness="55px" values={VALUES} />
      </Grommet>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('round', () => {
    const component = renderer.create(
      <Grommet>
        <Meter round values={VALUES} />
        <Meter type="circle" round values={VALUES} />
      </Grommet>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('background', () => {
    const component = renderer.create(
      <Grommet>
        <Meter background="light-3" values={VALUES} />
        <Meter background={{ color: 'light-3', opacity: 'medium' }} values={VALUES} />
        <Meter type="circle" background="light-3" values={VALUES} />
        <Meter
          type="circle"
          background={{ color: 'light-3', opacity: 'medium' }}
          values={VALUES}
        />
      </Grommet>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
