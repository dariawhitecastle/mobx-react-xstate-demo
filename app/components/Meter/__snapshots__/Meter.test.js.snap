// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meter background 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  max-width: 100%;
  overflow: hidden;
}

.c1 path {
  -webkit-transition: stroke 0.3s,stroke-width 0.3s;
  transition: stroke 0.3s,stroke-width 0.3s;
}

<div
  className="c0"
>
  <svg
    className="c1"
    height={24}
    preserveAspectRatio="none"
    viewBox="0 0 384 24"
    width={384}
  >
    <path
      d="M 0,12 L 384,12"
      fill="none"
      stroke="#EDEDED"
      strokeLinecap="square"
      strokeWidth={24}
    />
    <path
      d="M 0,12 L 76.8,12"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={24}
    />
  </svg>
  <svg
    className="c1"
    height={24}
    preserveAspectRatio="none"
    viewBox="0 0 384 24"
    width={384}
  >
    <path
      d="M 0,12 L 384,12"
      fill="none"
      stroke="#EDEDED"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={24}
    />
    <path
      d="M 0,12 L 76.8,12"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={24}
    />
  </svg>
  <svg
    className="c1"
    height={384}
    viewBox="0 0 384 384"
    width={384}
  >
    <circle
      cx={192}
      cy={192}
      fill="none"
      r={180}
      stroke="#EDEDED"
      strokeLinecap="square"
      strokeWidth={24}
    />
    <path
      d="M 363.1901729331 136.3769410125 A 180.0000000000 180.0000000000 0 0 0 192.0000000000 12.0000000000"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={24}
    />
  </svg>
  <svg
    className="c1"
    height={384}
    viewBox="0 0 384 384"
    width={384}
  >
    <circle
      cx={192}
      cy={192}
      fill="none"
      r={180}
      stroke="#EDEDED"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={24}
    />
    <path
      d="M 363.1901729331 136.3769410125 A 180.0000000000 180.0000000000 0 0 0 192.0000000000 12.0000000000"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={24}
    />
  </svg>
</div>
`;

exports[`Meter basic 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  max-width: 100%;
  overflow: hidden;
}

.c1 path {
  -webkit-transition: stroke 0.3s,stroke-width 0.3s;
  transition: stroke 0.3s,stroke-width 0.3s;
}

<div
  className="c0"
>
  <svg
    className="c1"
    height={24}
    preserveAspectRatio="none"
    viewBox="0 0 384 24"
    width={384}
  >
    <path
      d="M 0,12 L 384,12"
      fill="none"
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={24}
    />
    <path
      d="M 0,12 L 76.8,12"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={24}
    />
  </svg>
</div>
`;

exports[`Meter default 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  max-width: 100%;
  overflow: hidden;
}

.c1 path {
  -webkit-transition: stroke 0.3s,stroke-width 0.3s;
  transition: stroke 0.3s,stroke-width 0.3s;
}

<div
  className="c0"
>
  <svg
    className="c1"
    height={24}
    preserveAspectRatio="none"
    viewBox="0 0 384 24"
    width={384}
  >
    <path
      d="M 0,12 L 384,12"
      fill="none"
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={24}
    />
  </svg>
</div>
`;

exports[`Meter many values 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  max-width: 100%;
  overflow: hidden;
}

.c1 path {
  -webkit-transition: stroke 0.3s,stroke-width 0.3s;
  transition: stroke 0.3s,stroke-width 0.3s;
}

<div
  className="c0"
>
  <svg
    className="c1"
    height={24}
    preserveAspectRatio="none"
    viewBox="0 0 384 24"
    width={384}
  >
    <path
      d="M 0,12 L 384,12"
      fill="none"
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={24}
    />
    <path
      d="M 336,12 L 384,12"
      fill="none"
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={24}
    />
    <path
      d="M 288,12 L 336,12"
      fill="none"
      stroke="#00739D"
      strokeLinecap="butt"
      strokeWidth={24}
    />
    <path
      d="M 240,12 L 288,12"
      fill="none"
      stroke="#3D138D"
      strokeLinecap="butt"
      strokeWidth={24}
    />
    <path
      d="M 192,12 L 240,12"
      fill="none"
      stroke="#00873D"
      strokeLinecap="butt"
      strokeWidth={24}
    />
    <path
      d="M 144,12 L 192,12"
      fill="none"
      stroke="#A2423D"
      strokeLinecap="butt"
      strokeWidth={24}
    />
    <path
      d="M 96,12 L 144,12"
      fill="none"
      stroke="#00739D"
      strokeLinecap="butt"
      strokeWidth={24}
    />
    <path
      d="M 48,12 L 96,12"
      fill="none"
      stroke="#3D138D"
      strokeLinecap="butt"
      strokeWidth={24}
    />
    <path
      d="M 0,12 L 48,12"
      fill="none"
      stroke="#00873D"
      strokeLinecap="butt"
      strokeWidth={24}
    />
  </svg>
</div>
`;

exports[`Meter round 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  max-width: 100%;
  overflow: hidden;
}

.c2 path {
  -webkit-transition: stroke 0.3s,stroke-width 0.3s;
  transition: stroke 0.3s,stroke-width 0.3s;
}

.c1 {
  max-width: 100%;
  overflow: hidden;
  border-radius: 24px;
}

.c1 path {
  -webkit-transition: stroke 0.3s,stroke-width 0.3s;
  transition: stroke 0.3s,stroke-width 0.3s;
}

<div
  className="c0"
>
  <svg
    className="c1"
    height={24}
    preserveAspectRatio="none"
    viewBox="0 0 384 24"
    width={384}
  >
    <path
      d="M 12,12 L 372,12"
      fill="none"
      stroke="#F2F2F2"
      strokeLinecap="round"
      strokeOpacity="0.4"
      strokeWidth={24}
    />
    <path
      d="M 12,12 L 84,12"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="round"
      strokeWidth={24}
    />
  </svg>
  <svg
    className="c2"
    height={384}
    viewBox="0 0 384 384"
    width={384}
  >
    <circle
      cx={192}
      cy={192}
      fill="none"
      r={180}
      stroke="#F2F2F2"
      strokeLinecap="round"
      strokeOpacity="0.4"
      strokeWidth={24}
    />
    <path
      d="M 363.1901729331 136.3769410125 A 180.0000000000 180.0000000000 0 0 0 192.0000000000 12.0000000000"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="round"
      strokeWidth={24}
    />
    <path
      d="M 363.1901729331 136.3769410125 A 180.0000000000 180.0000000000 0 0 0 362.6982579371 134.8851618471"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="round"
      strokeWidth={24}
    />
  </svg>
</div>
`;

exports[`Meter size 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  max-width: 100%;
  overflow: hidden;
}

.c1 path {
  -webkit-transition: stroke 0.3s,stroke-width 0.3s;
  transition: stroke 0.3s,stroke-width 0.3s;
}

<div
  className="c0"
>
  <svg
    className="c1"
    height={24}
    preserveAspectRatio="none"
    viewBox="0 0 96 24"
    width={96}
  >
    <path
      d="M 0,12 L 96,12"
      fill="none"
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={24}
    />
    <path
      d="M 0,12 L 19.2,12"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={24}
    />
  </svg>
  <svg
    className="c1"
    height={24}
    preserveAspectRatio="none"
    viewBox="0 0 192 24"
    width={192}
  >
    <path
      d="M 0,12 L 192,12"
      fill="none"
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={24}
    />
    <path
      d="M 0,12 L 38.4,12"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={24}
    />
  </svg>
  <svg
    className="c1"
    height={24}
    preserveAspectRatio="none"
    viewBox="0 0 384 24"
    width={384}
  >
    <path
      d="M 0,12 L 384,12"
      fill="none"
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={24}
    />
    <path
      d="M 0,12 L 76.8,12"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={24}
    />
  </svg>
  <svg
    className="c1"
    height={24}
    preserveAspectRatio="none"
    viewBox="0 0 768 24"
    width={768}
  >
    <path
      d="M 0,12 L 768,12"
      fill="none"
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={24}
    />
    <path
      d="M 0,12 L 153.6,12"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={24}
    />
  </svg>
  <svg
    className="c1"
    height={24}
    preserveAspectRatio="none"
    viewBox="0 0 1152 24"
    width={1152}
  >
    <path
      d="M 0,12 L 1152,12"
      fill="none"
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={24}
    />
    <path
      d="M 0,12 L 230.4,12"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={24}
    />
  </svg>
  <svg
    className="c1"
    height={24}
    preserveAspectRatio="none"
    viewBox="0 0 24 24"
    width={24}
  >
    <path
      d="M 0,12 L 24,12"
      fill="none"
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={24}
    />
    <path
      d="M 0,12 L 4.8,12"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={24}
    />
  </svg>
  <svg
    className="c1"
    height={96}
    viewBox="0 0 96 96"
    width={96}
  >
    <circle
      cx={48}
      cy={48}
      fill="none"
      r={36}
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={24}
    />
    <path
      d="M 82.2380345866 36.8753882025 A 36.0000000000 36.0000000000 0 0 0 48.0000000000 12.0000000000"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={24}
    />
  </svg>
  <svg
    className="c1"
    height={192}
    viewBox="0 0 192 192"
    width={192}
  >
    <circle
      cx={96}
      cy={96}
      fill="none"
      r={84}
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={24}
    />
    <path
      d="M 175.8887473688 70.0425724725 A 84.0000000000 84.0000000000 0 0 0 96.0000000000 12.0000000000"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={24}
    />
  </svg>
  <svg
    className="c1"
    height={384}
    viewBox="0 0 384 384"
    width={384}
  >
    <circle
      cx={192}
      cy={192}
      fill="none"
      r={180}
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={24}
    />
    <path
      d="M 363.1901729331 136.3769410125 A 180.0000000000 180.0000000000 0 0 0 192.0000000000 12.0000000000"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={24}
    />
  </svg>
  <svg
    className="c1"
    height={768}
    viewBox="0 0 768 768"
    width={768}
  >
    <circle
      cx={384}
      cy={384}
      fill="none"
      r={372}
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={24}
    />
    <path
      d="M 737.7930240618 269.0456780925 A 372.0000000000 372.0000000000 0 0 0 384.0000000000 12.0000000000"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={24}
    />
  </svg>
  <svg
    className="c1"
    height={1152}
    viewBox="0 0 1152 1152"
    width={1152}
  >
    <circle
      cx={576}
      cy={576}
      fill="none"
      r={564}
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={24}
    />
    <path
      d="M 1112.3958751905 401.7144151725 A 564.0000000000 564.0000000000 0 0 0 576.0000000000 12.0000000000"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={24}
    />
  </svg>
  <svg
    className="c1"
    height={55}
    viewBox="0 0 55 55"
    width={55}
  >
    <circle
      cx={27.5}
      cy={27.5}
      fill="none"
      r={15.5}
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={24}
    />
    <path
      d="M 42.2413760026 22.7102365872 A 15.5000000000 15.5000000000 0 0 0 27.5000000000 12.0000000000"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={24}
    />
  </svg>
</div>
`;

exports[`Meter thickness 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  max-width: 100%;
  overflow: hidden;
}

.c1 path {
  -webkit-transition: stroke 0.3s,stroke-width 0.3s;
  transition: stroke 0.3s,stroke-width 0.3s;
}

<div
  className="c0"
>
  <svg
    className="c1"
    height={6}
    preserveAspectRatio="none"
    viewBox="0 0 384 6"
    width={384}
  >
    <path
      d="M 0,3 L 384,3"
      fill="none"
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={6}
    />
    <path
      d="M 0,3 L 76.8,3"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={6}
    />
  </svg>
  <svg
    className="c1"
    height={12}
    preserveAspectRatio="none"
    viewBox="0 0 384 12"
    width={384}
  >
    <path
      d="M 0,6 L 384,6"
      fill="none"
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={12}
    />
    <path
      d="M 0,6 L 76.8,6"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={12}
    />
  </svg>
  <svg
    className="c1"
    height={24}
    preserveAspectRatio="none"
    viewBox="0 0 384 24"
    width={384}
  >
    <path
      d="M 0,12 L 384,12"
      fill="none"
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={24}
    />
    <path
      d="M 0,12 L 76.8,12"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={24}
    />
  </svg>
  <svg
    className="c1"
    height={48}
    preserveAspectRatio="none"
    viewBox="0 0 384 48"
    width={384}
  >
    <path
      d="M 0,24 L 384,24"
      fill="none"
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={48}
    />
    <path
      d="M 0,24 L 76.8,24"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={48}
    />
  </svg>
  <svg
    className="c1"
    height={96}
    preserveAspectRatio="none"
    viewBox="0 0 384 96"
    width={384}
  >
    <path
      d="M 0,48 L 384,48"
      fill="none"
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={96}
    />
    <path
      d="M 0,48 L 76.8,48"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={96}
    />
  </svg>
  <svg
    className="c1"
    height={55}
    preserveAspectRatio="none"
    viewBox="0 0 384 55"
    width={384}
  >
    <path
      d="M 0,27.5 L 384,27.5"
      fill="none"
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={55}
    />
    <path
      d="M 0,27.5 L 76.8,27.5"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={55}
    />
  </svg>
  <svg
    className="c1"
    height={384}
    viewBox="0 0 384 384"
    width={384}
  >
    <circle
      cx={192}
      cy={192}
      fill="none"
      r={189}
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={6}
    />
    <path
      d="M 371.7496815798 133.5957880631 A 189.0000000000 189.0000000000 0 0 0 192.0000000000 3.0000000000"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={6}
    />
  </svg>
  <svg
    className="c1"
    height={384}
    viewBox="0 0 384 384"
    width={384}
  >
    <circle
      cx={192}
      cy={192}
      fill="none"
      r={186}
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={12}
    />
    <path
      d="M 368.8965120309 134.5228390463 A 186.0000000000 186.0000000000 0 0 0 192.0000000000 6.0000000000"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={12}
    />
  </svg>
  <svg
    className="c1"
    height={384}
    viewBox="0 0 384 384"
    width={384}
  >
    <circle
      cx={192}
      cy={192}
      fill="none"
      r={180}
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={24}
    />
    <path
      d="M 363.1901729331 136.3769410125 A 180.0000000000 180.0000000000 0 0 0 192.0000000000 12.0000000000"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={24}
    />
  </svg>
  <svg
    className="c1"
    height={384}
    viewBox="0 0 384 384"
    width={384}
  >
    <circle
      cx={192}
      cy={192}
      fill="none"
      r={168}
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={48}
    />
    <path
      d="M 351.7774947376 140.0851449450 A 168.0000000000 168.0000000000 0 0 0 192.0000000000 24.0000000000"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={48}
    />
  </svg>
  <svg
    className="c1"
    height={384}
    viewBox="0 0 384 384"
    width={384}
  >
    <circle
      cx={192}
      cy={192}
      fill="none"
      r={144}
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={96}
    />
    <path
      d="M 328.9521383465 147.5015528100 A 144.0000000000 144.0000000000 0 0 0 192.0000000000 48.0000000000"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={96}
    />
  </svg>
  <svg
    className="c1"
    height={384}
    viewBox="0 0 384 384"
    width={384}
  >
    <circle
      cx={192}
      cy={192}
      fill="none"
      r={164.5}
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={55}
    />
    <path
      d="M 348.4487969306 141.1667044253 A 164.5000000000 164.5000000000 0 0 0 192.0000000000 27.5000000000"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={55}
    />
  </svg>
</div>
`;

exports[`Meter type 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  max-width: 100%;
  overflow: hidden;
}

.c1 path {
  -webkit-transition: stroke 0.3s,stroke-width 0.3s;
  transition: stroke 0.3s,stroke-width 0.3s;
}

<div
  className="c0"
>
  <svg
    className="c1"
    height={24}
    preserveAspectRatio="none"
    viewBox="0 0 384 24"
    width={384}
  >
    <path
      d="M 0,12 L 384,12"
      fill="none"
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={24}
    />
    <path
      d="M 0,12 L 76.8,12"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={24}
    />
  </svg>
  <svg
    className="c1"
    height={384}
    viewBox="0 0 384 384"
    width={384}
  >
    <circle
      cx={192}
      cy={192}
      fill="none"
      r={180}
      stroke="#F2F2F2"
      strokeLinecap="square"
      strokeOpacity="0.4"
      strokeWidth={24}
    />
    <path
      d="M 363.1901729331 136.3769410125 A 180.0000000000 180.0000000000 0 0 0 192.0000000000 12.0000000000"
      fill="none"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeWidth={24}
    />
  </svg>
</div>
`;
